SDL 1.2 fluidsynth refactor

Wohlstand opened the pull request:
It's the similar refactoring I made at the mainstream branch with the SDL2.

sezero commented:
Thanks, applying this.

Although there are further things to fix here, will look shortly.

sezero commented:
Vitaly: I updated fluidsynth support in SDL-1.2 branch with further fixes.
If you have a chance, please test it (see the branch's history for the commits.)
Off-topic: Mp3 mad support is also updated there to fix a crash.

Wohlstand commented:
> Vitaly: I updated fluidsynth support in SDL-1.2 branch with further fixes.
> If you have a chance, please test it (see the branch's history for the commits.)

I'll try some after the sleep.

>  Off-topic: Mp3 mad support is also updated there to fix a crash.

:fox_face: :+1:
